### DISCUSSION TITLE
Signals or Hooks

### DISCUSSION THREAD

**Discussion by Fidel-C** [piccolo-orm/piccolo#882] [General]
Do we or can we have the equivalent of Django signals in piccolo ORM? That is a code that runs after or before a given action in the database. Such as when an item is created, updated or deleted. This is a very useful feature.

**Comment by dantownsend** [comment]
We don't currently have signals. We have hooks in Piccolo Admin which help in some situations, but having some kind of signals built into Piccolo itself would be nice.

**Comment by Fidel-C** [comment]
> We don't currently have signals. We have hooks in Piccolo Admin which help in some situations, but having some kind of signals built into Piccolo itself would be nice.

Thanks for your response.
